bin/manager_start.sh

```diff
@@ -27,26 +23,46 @@ JAVA_OPTS="-XX:MaxDirectMemorySize=$MAX_DIRECT_MEMORY
 
 MAIN="spark.jobserver.JobManager"
 
-MESOS_OPTS=""
-if [ $1 == "mesos-cluster" ]; then
-    MESOS_OPTS="--master $MESOS_SPARK_DISPATCHER --deploy-mode cluster"
-    appdir=$REMOTE_JOBSERVER_DIR
-fi
+# copy files via spark-submit and read them from current (container) dir
+if [ $2 = "cluster" -a -z "$REMOTE_JOBSERVER_DIR" ]; then
+  SPARK_SUBMIT_OPTIONS="$SPARK_SUBMIT_OPTIONS
+    --master $1 --deploy-mode cluster
+    --conf spark.yarn.submit.waitAppCompletion=false
+    --files $appdir/log4j-cluster.properties,$conffile"
+  JAR_FILE="$appdir/spark-job-server.jar"
+  CONF_FILE=$(basename $conffile)
+  LOGGING_OPTS="-Dlog4j.configuration=log4j-cluster.properties"
 
-if [ ! -z $5 ]; then
-  cmd='$SPARK_HOME/bin/spark-submit --class $MAIN --driver-memory $JOBSERVER_MEMORY
-  --conf "spark.executor.extraJavaOptions=$LOGGING_OPTS"
-  --proxy-user $5
-  $MESOS_OPTS
-  --driver-java-options "$GC_OPTS $JAVA_OPTS $LOGGING_OPTS $CONFIG_OVERRIDES"
-  $appdir/spark-job-server.jar $2 $3 $4 $conffile'
+# use files in REMOTE_JOBSERVER_DIR
+elif [ $2 == "cluster" ]; then
+  SPARK_SUBMIT_OPTIONS="$SPARK_SUBMIT_OPTIONS
+    --master $1 --deploy-mode cluster
+    --conf spark.yarn.submit.waitAppCompletion=false"
+  JAR_FILE="$REMOTE_JOBSERVER_DIR/spark-job-server.jar"
+  CONF_FILE="$REMOTE_JOBSERVER_DIR/$(basename $conffile)"
+  LOGGING_OPTS="-Dlog4j.configuration=$REMOTE_JOBSERVER_DIR/log4j-cluster.properties"
+
+# client mode, use files from app dir
 else
-  cmd='$SPARK_HOME/bin/spark-submit --class $MAIN --driver-memory $JOBSERVER_MEMORY
-  --conf "spark.executor.extraJavaOptions=$LOGGING_OPTS"
-  --driver-java-options "$GC_OPTS $JAVA_OPTS $LOGGING_OPTS $CONFIG_OVERRIDES"
-  $MESOS_OPTS
-  $appdir/spark-job-server.jar $2 $3 $4 $conffile'
+  JAR_FILE="$appdir/spark-job-server.jar"
+  CONF_FILE="$conffile"
+  LOGGING_OPTS="-Dlog4j.configuration=file:$appdir/log4j-server.properties -DLOG_DIR=$5"
+  GC_OPTS="$GC_OPTS -Xloggc:$5/gc.out"
+fi
+
+if [ -n "$6" ]; then
+  SPARK_SUBMIT_OPTIONS="$SPARK_SUBMIT_OPTIONS --proxy-user $6"
+fi
+
+if [ -n "$JOBSERVER_KEYTAB" ]; then
+  SPARK_SUBMIT_OPTIONS="$SPARK_SUBMIT_OPTIONS --keytab $JOBSERVER_KEYTAB"
 fi
 
-eval $cmd
+cmd='$SPARK_HOME/bin/spark-submit --class $MAIN --driver-memory $JOBSERVER_MEMORY
+      --conf "spark.executor.extraJavaOptions=$LOGGING_OPTS"
+      $SPARK_SUBMIT_OPTIONS
+      --driver-java-options "$GC_OPTS $JAVA_OPTS $LOGGING_OPTS $CONFIG_OVERRIDES $SPARK_SUBMIT_JAVA_OPTIONS"
+      $JAR_FILE $3 $4 $CONF_FILE'
+
+eval $cmd 2>&1 > $5/spark-job-server.out
```

Comment:
The yarn cluster mode client does not read the log4j properties and logs to stdout... All errors before this line are going into the global `spark-job-server.log` and all output of spark-submit  into its context specific log.